Update README.md and add Progress bar

ekinyuksel12 opened the pull request.

ekinyuksel12 committed:
Add optional progress bar

I've added a few lines of code to create a download progress bar to
track the progress of download. You can use the progress bar with
'--progress' argument

ekinyuksel12 committed:
Update Readme

This commit updates the readme file to add a description and a little
part to show the usage of '--progress'

ekinyuksel12 committed:
Update Readme for the new fork

This commit updates the readme for making it a fork repo

pakoito commented:
I understand what this is doing and why, and I like it. But I'd rather not bring an extra dependency in a PR, specially one that's not been updated in 2 years. Is there any other way of implementing this, even if it's less nice?

ekinyuksel12 commented:
I can take look in a few days. I think it is possible without dependencies. But it was easier with a library so that's I did chose that. I will take look and update you. 

Do I need to create another PR or can I update the commit here? That's my first time contributing.

pakoito commented:
Updating this one should be enough, thanks! You'd need to update the README to remove the reference that this is a fork, as it'll go into the original repo :D

ekinyuksel12 commented:
Sure thing. I only forked the project because I guess you disabled commits from strangers. I will try to take a look today. Thanks

pakoito commented:
It's the default on github, you fork, make changes and submit a PR upstream :D You can keep the readme changes in the repo, cherrypick them out of the PR.

ekinyuksel12 committed:
Remove cli-progress dependency

ekinyuksel12 committed:
Update README to remove fork note

ekinyuksel12 committed:
Update dependency locks

ekinyuksel12 commented:
I've updated the code. I removed the fork notice and cli-progress library dependency. It made the code a little more verbose but it works like the actual library. Codebase is dirtier than my liking but I will take look at that later. Maybe in another pull request. Thanks for showing me how to properly PR. Let me know if this needs any more updates.

pakoito commented on index.js:
```diff
@@ -3,105 +3,175 @@ import got from 'got';
+        .map(({
+            Name,
+            Id
+        }) => [
```
Not a blocker, just generic perf advice: filter followed by map can be done in a single pass using flatMap.

```typescript
.flatMap({
  /*clause*/
    ? [/*map*/] 
    : []
})
```

pakoito commented:
Approved and merged. Thanks!
